KubeArmor/core/kubeArmor.go

```diff
@@ -446,12 +451,44 @@ func KubeArmor() {
 		}
 	}
 
-	// == //
+	trimmedSocket := trimmedSocketPath()
+
+	// Un-orchestrated workloads
+	if !dm.K8sEnabled && cfg.GlobalCfg.Policy {
+
+		// Check if cri socket set, if not then auto detect
+		if cfg.GlobalCfg.CRISocket == "" {
+			if kl.GetCRISocket("") == "" {
+				dm.Logger.Warnf("Error while looking for CRI socket file")
+			} else {
+				cfg.GlobalCfg.CRISocket = "unix://" + kl.GetCRISocket("")
+				// update the value of trimmed socket path when the cfg.GlobalCfg.CRISocket is not set
+				trimmedSocket = trimmedSocketPath()
```

Comment:
What is the use of trimmedSocket below?
All we need is a flag saying `enableContainerMonitoring := true` and use it in place of `os.Stat(trimmedSocket)` below. Note that `GetCRISocket()` returns successfully only when the path is available. Hence setting a flag should be enough.

Reply:
trimmedSocket := trimmedSocketPath() on line 454 checks if the sock file exist, but if the GlobalCfg.CRISocket is not set then we are setting it here, but the trimmedSocket value set above won't be updated, that's why it's called here to reflect the updated value of CRISocket flag.